osidb/api_views.py

```diff
@@ -83,7 +84,92 @@ def healthy(request: Request) -> Response:
     return Response()
 
 
-class StatusView(APIView):
+class RudimentaryUserPathLoggingMixin:
+    # Provides rudimentary visibility into API requests, users, time of processing via logging.
+    # This is a stop-gap solution and not fit for enhancement.
+    # Logs must be sent and stored securely w.r.t. the data sent there:
+    # - username
+    # - HTTP method and path
+    # - how long the request took
+    # - time of event (added implicitly by the logger)
+    # (This is not a docstring so as not to pollute the generated schema.)
+
+    def initialize_request(self, request, *args, **kwargs):
+        """
+        Log beginning of API request.
+        """
+
+        request = super().initialize_request(request, *args, **kwargs)
+
+        if getattr(self, "swagger_fake_view", False):
+            return request
+
+        try:
+            # Different sources suggest different behavior of request.user.
+            # Local testing shows that str(request.user) is always nonempty
+            # (AnonymousUser or the username).
+            # However, as this is not fully explored and probably not worth it,
+            # allow seeing unexpected states in the logs.
+            if request.user and request.user.is_authenticated and request.user.username:
+                user = f"USER:{request.user.username}"
+            elif request.user and request.user.is_authenticated:
+                # Probably shouldn't happen.
+                user = f"USER_AUTH_NOUSERNAME:{str(request.user)}"
+            elif request.user and request.user.username:
+                user = f"USER_UNAUTH:{request.user.username}"
+            elif request.user:
+                user = f"USER_UNAUTH_NOUSERNAME:{str(request.user)}"
+            else:
+                user = f"USER_NONE:{repr(request.user)}"
+        except:  # noqa: E722 Using bare except on purpose
+            # Probably only happens when generating schema,
+            # preempting that using getattr(self, "swagger_fake_view", False)
+            # is much more performant, but leaving this try/except
+            # so that the logging doesn't break normal operation
+            # in an unexpected cornercase (not enough time to investigate deeply).
+            user = "USER_EXCEPTION"
+
+        method = request.method.upper()
+        path = request.path
+
+        request._rudimentary_user_path_logging = {
+            "method": method,
+            "path": path,
+            "start": datetime.now(),
+            "user": user,
+        }
+
+        log_message = f"API REQUEST {user} at {method} {path}"
+        logger.debug(log_message)
```

Comment:
Is `debug` level enough? Shouldn't it be maybe `info`? The production (where it matters) does not run in the debug mode so I expect that this logging will have no effect there. Or do miss something?

Reply:
Good question. Honestly, I have very little idea how that works.  I see there's some configuration in `settings_prod.py` and based on that and Splunk it seems prod-django.log (which would most probably be the log where these API request logging messages would go to) is set to WARNING level, whereas everything else seems to have (implicit?) INFO. I'll study this and ask around the team to understand it better. Thank you!

Reply:
If you find out that the log level is not set appropriately, it is a viable approach to just adjust it. WARNING level seems to me too strict, I would expect INFO level. However, DEBUG level on the other hand seems to be to unrestricted. I would consider debug logs as something necessary for debugging so not something one would expect to be used in production.

Reply:
Thanks, that's partly what I did in the end.

After learning a bit about logging, it seems it would be too complicated to have e.g. 'osidb' module at level WARNING and nested 'osidb.api_views' module at level INFO (seems Python likes it in the opposite order of levels, which doesn't fit our usecase).

So I added a separate logging module 'api_req' and modified the 'console' level for prod.

Here's the change for easier review: https://github.com/RedHatProductSecurity/osidb/compare/f66d43619d22034b36d57befd367961e83c0c017..ecab2150f3b929d6b244a6547ed24d0af4dc8223